src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1789,10 +1950,21 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
 
         // check for loan restricting path P being used. Accounts for
         // borrows of P, P.a.b, etc.
-        for i in flow_state.borrows.elems_incoming() {
-            let borrowed = &data[i];
+        let mut elems_incoming = flow_state.borrows.elems_incoming();
+        while let Some(i) = elems_incoming.next() {
+            // Skip any reservation that has a corresponding current
+            // activation.  This way, the traversal will visit each
+            // borrow_index at most once.
+            if let Some(j) = elems_incoming.peek() {
```

Comment:
Is this needed? Don't we already make sure only 1 error is emitted even if the reservation/activation come from separate instructions?

Reply:
ok i removed it.